packages/workspace/src/generators/new/generate-workspace-files.ts

```diff
@@ -51,6 +51,10 @@ function setPresetProperty(tree: Tree, options: NormalizedSchema) {
       delete json.targetDefaults;
       delete json.workspaceLayout;
       delete json.npmScope;
+      json.workspaceLayout = {
```

Comment:
This was removed because it should be automatic as long as the directory exists, which it does for the TS preset.

Reply:
i was trying to fix https://github.com/nrwl/nx/issues/16391 about react-native with npm preset. 
`workspaceLayout().libsDir` always returns libs
i realize that for workspaceLayout() function from devkit just returns whatever in nx.json. so i create another pr https://github.com/nrwl/nx/pull/16624/files#diff-414859da125ccd6aa42020f685ec3a2a647b3b20d3333ba73afc71aa02e8700a.